pandas/core/indexes/datetimes.py

```diff
@@ -283,10 +283,19 @@ def __new__(cls, data=None,
 
         # By this point we are assured to have either a numpy array or Index
         data, copy = maybe_convert_dtype(data, copy)
-        if not (is_datetime64_dtype(data) or is_datetime64tz_dtype(data) or
-                is_integer_dtype(data) or lib.infer_dtype(data) == 'integer'):
-            data = tools.to_datetime(data, dayfirst=dayfirst,
-                                     yearfirst=yearfirst)
+
+        if is_object_dtype(data) or is_string_dtype(data):
+            # TODO: We do not have tests specific to string-dtypes,
```

Comment:
you *could* just write this as

```
try:
      data = datas.astype(np.int64)
except:
     ...
```
*might* be more clear

Reply:
The issue with that is np.array(['20160405']) becomes np.array([20160405]) instead of 2016-04-05.

Reply:
ok sure